web/src/app/admin/AdminDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import List from '@mui/material/List'
+import ListItem from '@mui/material/ListItem'
+import ListItemText from '@mui/material/ListItemText'
+import Typography from '@mui/material/Typography'
+import { omit } from 'lodash'
+import FormDialog from '../dialogs/FormDialog'
+import { nonFieldErrors, fieldErrors } from '../util/errutil'
+import Diff from '../util/Diff'
+import {
+  DocumentNode,
+  OperationVariables,
+  TypedDocumentNode,
+  useMutation,
+} from '@apollo/client'
+
+interface Value {
+  id: string
+  oldValue: number
+  value: number
+  type: string
+}
+
+interface FieldValues {
+  [id: string]: string
+}
+
+interface AdminDialogProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  mutation: DocumentNode | TypedDocumentNode<any, OperationVariables>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  values: Value[] | any
+  fieldValues: FieldValues
+  onClose: () => void
+  onComplete: () => void
+}
+
+function AdminDialog(props: AdminDialogProps): JSX.Element {
+  const [commit, { error }] = useMutation(props.mutation, {
+    onCompleted: props.onComplete,
+  })
+  const changeKeys = Object.keys(props.fieldValues)
+  const changes = props.values
+    .filter((v: { id: string }) => changeKeys.includes(v.id))
+    .map((orig: { id: string | number; value: number; type: string }) => ({
```

Comment:
We can remove the type here once we set the above to `ConfigValue[]`

```suggestion
    .map((orig) => ({
```